File: comparisons/parity/glmm_offset.jl
# Poisson GLMM with offset parity exporter (MixedModels.jl).
#   julia comparisons/parity/glmm_offset.jl

using CSV
using DataFrames
using JSON
using MixedModels

file_path = joinpath("tests", "data", "grouseticks.csv")
isfile(file_path) || error("missing dataset: $file_path")

df = CSV.read(file_path, DataFrame)
df.BROOD = string.(df.BROOD)

form = @formula(TICKS ~ 1 + YEAR96 + YEAR97 + offset(log_height) + (1 | BROOD))
m = fit(MixedModel, form, df, Poisson(), LogLink())

coef_names = coefnames(m)
coef_vals = coef(m)
coefficients = Dict(zip(coef_names, coef_vals))

payload = Dict(
    "case" => "glmm_offset",
    "implementation" => "julia",
    "coefficients" => coefficients,
)
println(JSON.json(payload))

File: comparisons/parity/glmm_offset.R
source("comparisons/parity/parity_r_setup.R")
suppressPackageStartupMessages({
  library(lme4)
  library(jsonlite)
})
source("comparisons/r_setup.R")

data <- read.csv("tests/data/grouseticks.csv")
data$BROOD <- as.factor(data$BROOD)
fit <- glmer(
  TICKS ~ YEAR96 + YEAR97 + offset(log_height) + (1 | BROOD),
  data = data,
  family = poisson(link = "log")
)
coefs <- as.list(as.numeric(fixef(fit)))
names(coefs) <- names(fixef(fit))
cat(toJSON(
  list(case = "glmm_offset", implementation = "r", coefficients = coefs),
  auto_unbox = TRUE,
  digits = 10
))
